msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java

```diff
@@ -1914,4 +2040,225 @@ else if (toolId.equals(DiscussionForumService.MESSAGES_TOOL_ID))
   	return eventMessagePrefix + contextId + "/" + object.toString() + "/" + userId;
   }
 
+  public PrivateMessage getPrivateMessage(final String id) {
+	  PrivateMessage currentMessage = (PrivateMessage) messageManager.getMessageByIdWithAttachments(Long.parseLong(decrypt(id)));
+	  getHibernateTemplate().initialize(currentMessage.getRecipients());
+	  return currentMessage;
+  }
+  
+  public PrivateMessage getPvtMsgReplyMessage(PrivateMessage currentMessage, MimeMessage msg, StringBuilder[] bodyBuf, List<Reference> attachments, String from) throws MessagingException {
+		
+	  PrivateMessage rrepMsg = messageManager.createPrivateMessage() ;
+	  try {
+		  rrepMsg.setTitle(msg.getSubject());
+		  try {
+			  rrepMsg.setCreatedBy(userDirectoryService.getUserByEid(from).getId());
+		  } catch (UserNotDefinedException e) {
+			  rrepMsg.setCreatedBy(getUserIdByFowardMail(currentMessage.getId(), from));
+			  if (rrepMsg.getCreatedBy() == null) {
+				  log.warn("PrivateMessageManagerImpl.getPvtMsgReplyMessage: exception: 683 - Sender's email does not belong in Sakai, nor is it recognized as an automatic forwarding address. If you have replied to this email from an address other than the one you have configured for email forwarding, please reply from that address.");
+				  throw new MessagingException("683 - Sender's email does not belong in Sakai, nor is it recognized as an automatic forwarding address. If you have replied to this email from an address other than the one you have configured for email forwarding, please reply from that address."); 
+			  }
+		  }
+		  User userFrom;
+		  try {
+			  userFrom = userDirectoryService.getUser(rrepMsg.getCreatedBy());
+			  from = userFrom.getEmail();
+		  } catch (UserNotDefinedException e1) {
+			  log.warn("PrivateMessageManagerImpl.getPvtMsgReplyMessage: exception: 683 - Sender's email does not belong in Sakai, nor is it recognized as an automatic forwarding address. If you have replied to this email from an address other than the one you have configured for email forwarding, please reply from that address.");
```

Comment:
no need to add the method identifier PrivateMessageManagerImpl.getPvtMsgReplyMessage as the logger does this already.